public/app/features/dashboard/api/publicDashboardApi.ts

```diff
@@ -133,6 +178,38 @@ export const publicDashboardApi = createApi({
         'AuditTablePublicDashboard',
       ],
     }),
+    updatePublicDashboardAccess: builder.mutation<
+      PublicDashboard,
+      {
+        dashboard: (Pick<DashboardModel, 'uid'> & Partial<Pick<DashboardModel, 'updateMeta'>>) | DashboardScene;
+        payload: Partial<PublicDashboard>;
+      }
+    >({
+      query: ({ payload, dashboard }) => {
+        const dashUid = dashboard instanceof DashboardScene ? dashboard.state.uid : dashboard.uid;
+        return {
+          url: `/dashboards/uid/${dashUid}/public-dashboards/${payload.uid}`,
+          method: 'PATCH',
+          data: payload,
+        };
+      },
+      async onQueryStarted({ dashboard, payload: { share } }, { dispatch, queryFulfilled }) {
+        await queryFulfilled;
+        let message = 'Public dashboard updated!';
+
+        if (config.featureToggles.newDashboardSharingComponent) {
+          message =
+            share === PublicDashboardShareType.PUBLIC
+              ? 'Dashboard access updated: Anyone with the link can now access'
```

Comment:
In general, I found several messages without translation. Are we planning to do it later, or should it be part of this PR?

Reply:
sorry, I missed these messages. However, once we merge this PR we will need help of @imatwawana to review all the texts

Reply:
Maybe we want to wait until we have the final text so we don't translate them twice? In that case, we can take care of this in another PR.

Reply:
There's no "second" translation. I'll do that in this PR so we have the <Trans> component added in these cases. THen, we can change the text and that's all